eras/conway/impl/cddl-files/conway.cddl

```diff
@@ -427,18 +427,31 @@ drep_voting_thresholds =
   ]
 
 transaction_witness_set =
-  { ? 0: [* vkeywitness ]
-  , ? 1: [* native_script ]
-  , ? 2: [* bootstrap_witness ]
-  , ? 3: [* plutus_v1_script ]
-  , ? 4: [* plutus_data ]
-  , ? 5: [* redeemer ]
-  , ? 6: [* plutus_v2_script ]
-  , ? 7: [* plutus_v3_script ]
+  { ? 0: [+ vkeywitness ]
+  , ? 1: [+ native_script ]
+  , ? 2: [+ bootstrap_witness ]
+  , ? 3: plutus_v1_scripts
+  , ? 4: [+ plutus_data ]
+  , ? 5: [+ redeemer ]
+  , ? 6: plutus_v2_scripts
+  , ? 7: plutus_v3_scripts
   }
 
+; The real type of plutus scripts fields (3, 6, 7) is nonempty_set<bytes>.
+; However, because the cddl tool we use for roundtrip testing doesn't generate distinct collections,
+; we are defining them here as `distinct_nonempty_set_or_list`,
+; which guarantees that the generated collection is nonempty and distinct
+
+; plutus_v1_scripts = nonempty_set<plutus_v1_script>
+plutus_v1_scripts = distinct_nonempty_set_or_list<plutus_v1_script>
 plutus_v1_script = bytes
+
+; plutus_v2_scripts = nonempty_set<plutus_v2_script>
+plutus_v2_scripts = distinct_nonempty_set_or_list<plutus_v2_script>
 plutus_v2_script = bytes
+
+; plutus_v3_scripts = nonempty_set<plutus_v3_script>
+plutus_v3_scripts = distinct_nonempty_set_or_list<plutus_v3_script>
 plutus_v3_script = bytes
```

Comment:
That would be awesome, but I think in order for this to work,  the definition of `nonempty_set` would have to be this:
`nonempty_set<a> = #6.258(a) / a`
instead of  what it is, namely:
`nonempty_set<a> = #6.258([+ a]) / [+ a]` . 

With your suggestion, there is an extra list in `plutus_v1_scripts` , from the need to define `distinct_bytes` as a  list:
`nonempty_set<plutus_script>` = ` #6.258([+ plutus_script]) / [+ plutus_script]` =  ` #6.258([+ distinct_bytes<bytes>]) / [+ distinct_bytes<bytes>]`  = ` #6.258([+ [bytes, bytes... ]) / [+ [bytes, bytes ...]]`

Reply:
As discussed, this problem goes away if `distinct` is replaced with `a .size 8 / a.size 16 ...`  (instead of the list). 
Thanks!